app/controllers/display_ads_controller.rb

```diff
@@ -0,0 +1,30 @@
+class DisplayAdsController < ApplicationController
+  before_action :set_cache_control_headers, only: %i[for_display], unless: -> { current_user }
+  CACHE_EXPIRY_FOR_DISPLAY_ADS = 15.minutes.to_i.freeze
+
+  def for_display
+    skip_authorization
+    set_cache_control_headers(CACHE_EXPIRY_FOR_DISPLAY_ADS) unless session_current_user_id
```

Comment:
Is this redundant with the above `before_action`? Probably not technically a problem if so, but maybe we don't need to have it in both places?

Reply:
I thought so too initially. However, we seem to set the cache header in the `before_action` and then override the max age by calling the method again the action itself in all the places where we override the default expiry time:

Some examples:
Articles Controller:
- https://github.com/forem/forem/blob/f88a8611251160adff772994f2a4c6bdb7595c60/app/controllers/articles_controller.rb#L55
- https://github.com/forem/forem/blob/f88a8611251160adff772994f2a4c6bdb7595c60/app/controllers/articles_controller.rb#L10

Tagged Articles controllre:
- https://github.com/forem/forem/blob/f88a8611251160adff772994f2a4c6bdb7595c60/app/controllers/stories/tagged_articles_controller.rb#L3
- https://github.com/forem/forem/blob/f88a8611251160adff772994f2a4c6bdb7595c60/app/controllers/stories/tagged_articles_controller.rb#L26-L28

and it also in the Sitemaps Controller, and the Sourcemap controller.

 I'll look into this further to see if its necessary, on the surface it does not look to be.

Reply:
Just dropping a note here that I'll create a follow up PR to address this after further investigation.